src/Razor/src/Microsoft.VisualStudio.Editor.Razor/BackgroundDocumentGenerator.cs

```diff
@@ -170,7 +178,8 @@ public void Enqueue(ProjectSnapshot project, DocumentSnapshot document)
 
             lock (_work)
             {
-                if (_projectManager.IsDocumentOpen(document.FilePath))
+                if (_projectManager.IsDocumentOpen(document.FilePath)
+                        && (!forceInitialDocumentGeneration || !_lspEditorFeatureDetector.IsLSPEditorFeatureEnabled()))
```

Comment:
Why do we need to check the feature detector piece here or have the force initial document generation?

Reply:
As it stands, we never actually generate the document at all until it's closed. Consequently, we don't have symbols / compilation / taghelpers for the new component until the document is closed.

I don't think this background document generation should lead to conflicts as it is triggered only on the document creation event and we are in an LSP enabled editor. Please let me know if you believe this to be problematic, in which case we'll need to find another way to do initial document generation.

Reply:
Discussed offline. The [`UpdateLSPFileInfo` function](https://github.com/dotnet/aspnetcore-tooling/blob/f916e22a27cdf7453757c08503d6097805bd2ce4/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorDynamicFileInfoProvider.cs#L45) should be leading to document generation on open (https://github.com/dotnet/aspnetcore-tooling/blob/4cfabbc2072339506349dbaae8e6ba17f3b261a8/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs#L163) however the `OpenDocument` function isn't being called. This will be addressed separately.